ndonnx/_core/_impl.py

```diff
@@ -521,6 +521,14 @@ def where_dtype_agnostic(a: ndx.Array, b: ndx.Array) -> ndx.Array:
                         b.astype(dtypes.int32)._core(),
                     )
                 )
+            elif a.dtype in (dtypes.uint16, dtypes.uint32, dtypes.uint64):
```

Comment:
```suggestion
            elif a.dtype in (dtypes.uint16, dtypes.uint32):
```

Reply:
I think in this case it's actually safe. There is no loss, the onnx standard defines cast between signed and unsigned as just reinterpreting the bits. It is only unsafe for arithmetic